### ISSUE TITLE
[css-color-4] Incorrect description on preprocessing hue angles for interpolation

### ISSUE THREAD

**Issue by yarusome** [w3c/csswg-drafts#8838]
The [Hue Interpolation](https://drafts.csswg.org/css-color/#hue-interpolation) section says:

> Both angles and their difference need to be constrained to [0, 360] prior to interpolation. To do this, the minimum number of turns that fit in the lesser angle is added or subtracted from both angles, bringing the lesser angle into the range [0,360); and if the difference between them is greater than 360° then the minimum number of turns to bring the difference into the range [0,360] is further subtracted from the greater angle.

However, the angle pair <code>(θ<sub>1</sub>, θ<sub>2</sub>) = (40deg, 400deg)</code> passes the preprocessing steps without being modified, leaving with <code>θ<sub>2</sub> > 360deg</code>.

Based on the four algorithms described in this section, it seems that only the lesser of the two angles needs to be in `[0deg, 360deg)` (and the difference be in `[0deg, 360deg]`).

**Comment by svgeesus** [comment]
@LeaVerou could you look into this please?

**Comment by svgeesus** [comment]
@LeaVerou I suspect that should be (my emphasis)

> constrained to [0, 360 **)**

and

> is greater than **or equal to** 360° 

which, in the example above, would give 400 - 360 = 40

**Comment by yarusome** [comment]
@svgeesus Two small issues regarding 2641c47:

1. After this commit, it's no longer possible to specify a full circumference with `increasing hue` and `decreasing hue`. Is this behavioral change intended? (I'd like to make sure before updating MDN.)
2. The explanatory texts for pseudo-codes:

   https://github.com/w3c/csswg-drafts/blob/4da3c11f4e7be9aff72711a576e8efc1edf2c75e/css-color-4/Overview.bs#L4893-L4912

   may also be updated:

   ```diff
   - 	Angles are adjusted so that |θ₂ - θ₁| ∈ [0, 360]. In pseudo-Javascript:
   + 	Angles are adjusted so that |θ₂ - θ₁| ∈ [0, 360). In pseudo-Javascript:
   - 	Angles are adjusted so that |θ₂ - θ₁| ∈ [-360, 0]. In pseudo-Javascript:
   + 	Angles are adjusted so that |θ₂ - θ₁| ∈ (-360, 0]. In pseudo-Javascript:
   ```

**Comment by svgeesus** [comment]
> it's no longer possible to specify a full circumference

Yes, intentional. We actually decided that a while ago (when we changed HSL to allow round-tripping of out of gamut colors) and I noticed that I had not updated the spec there.

Thanks! Will update the pseudo-codes asap.

**Comment by yarusome** [comment]
@svgeesus Sorry to bother again :pray:, but is `<hue>` also supposed to be normalized into `[0, 360)` now? If that's the case, then two more places need edits:

https://github.com/w3c/csswg-drafts/blob/0645f1bfda3d5e1a9b17adf50c175914eec3999d/css-color-4/Overview.bs#L971-L981

https://github.com/w3c/csswg-drafts/blob/0645f1bfda3d5e1a9b17adf50c175914eec3999d/css-color-4/Overview.bs#L6653-L6659

**Comment by svgeesus** [comment]
Well spotted, agreed and fixed for the first one. Te second is in a change log so it stays (but there will be another entry in the changes, when I add recent changes, about this).
